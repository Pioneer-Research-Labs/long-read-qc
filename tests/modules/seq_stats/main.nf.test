nextflow_process {

    name "Test Process seq_stats"
    script "modules/seq_stats/main.nf"
    process "seq_stats"

    test("Should generate sequencing stats") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple([ id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file("${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz"),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/inserts.fasta'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/barcodes.fasta'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("${launchDir}/results/Gateway_EmptyBarcodedLib1_20241217/seq_stats.tsv").match()
        }

    }

}
