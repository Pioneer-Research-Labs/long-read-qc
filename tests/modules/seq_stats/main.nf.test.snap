{
    "Should generate sequencing stats": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/1ca178391e18368ee6032227fb2fcbcb/results/Gateway_EmptyBarcodedLib1_20241217/seq_stats.tsv"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-01T13:38:08.208524"
    }
}